core/src/test/scala/integration/kafka/tools/DumpLogSegmentsIntegrationTest.scala

```diff
@@ -0,0 +1,85 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package integration.kafka.tools
+
+import java.io.{ByteArrayOutputStream, PrintStream}
+import java.util.Properties
+
+import kafka.integration.KafkaServerTestHarness
+import kafka.log.Log
+import kafka.server.KafkaConfig
+import kafka.tools.DumpLogSegments
+import kafka.utils.TestUtils
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerConfig, ProducerRecord}
+import org.apache.kafka.common.serialization.ByteArraySerializer
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+/**
+  * Testing of [[DumpLogSegments]] tool
+  */
+class DumpLogSegmentsIntegrationTest extends KafkaServerTestHarness {
```

Comment:
Curious if we really need a broker for this. Maybe we could just construct a `Log` and write directly to it?